### PR TITLE
[FEAT] moving from basic hash for message to a struct with some helpe…

### PR DESCRIPTION
**Pull request by nitsanfiverr**
…rs to get the right type of values

### CODE REVIEW

File: worker/rabbit.go [hunk 1]
```diff
@@ -37,38 +81,39 @@ func Consume(queueName string) {
...
+	if !event.Valid {
+		return errors.New("Event is not valid")
+	}
 
+	if event.Name == "" {
+		return errors.New("Event name is empty")
 	}
-	return errors.Newf("Couldn't find event for params %v", p)
+	//check if there is valid task for the event
+	if task, found := tasksMapping[event.Name]; found {
```
Comment (roeepolegfiverr) [inline worker/rabbit.go]: i think it's better to do validation first
````
if task, found := tasksMapping[event.Name]; found == nil {
  return errors.Newf("Couldn't find task for event %+v", event)
}
return task(event)
````

File: worker/rabbit.go [hunk 2]
```diff
@@ -37,38 +81,39 @@ func Consume(queueName string) {
 		}
 		fmt.Printf("Worker starting on queue %s\nWaiting for some messages to work on\n", queueName)
 
-		for message := range messages {
-			fmt.Printf("got message with body:%s \n", message.Body)
-			options, err := parseMessage(message)
+		for delivery := range deliveries {
```
Comment (roeepolegfiverr) [inline worker/rabbit.go]: [NAMING] i prefer message over delivery

File: worker/rabbit.go [hunk 3]
```diff
@@ -79,26 +124,41 @@ func ackMessage(fn func() error, message amqp.Delivery) {
 	}
 }
 
-func parseMessage(message amqp.Delivery) (map[string]interface{}, error) {
+func parseDelivery(delivery amqp.Delivery) (*Event, error) {
```
Comment (roeepolegfiverr) [inline worker/rabbit.go]: it's basically and Event init method, should it be on the event struct?

File: worker/rabbit.go (hunk 2 above)
Comment (nitsanfiverr) [inline worker/rabbit.go]: it's how rabbit calls it (amqp.Delivery). I use eventMessage, so another message might be confusing

File: worker/rabbit.go (hunk 3 above)
Comment (nitsanfiverr) [inline worker/rabbit.go]: I'll try it.

Comment (nitsanfiverr) [inline worker/rabbit.go]: well, it's not right. try to init nil object with it's own method.

File: worker/rabbit.go [hunk 4]
```diff
@@ -13,7 +13,63 @@ import (
...
-type params map[string]interface{}
+type Event struct {
+	Params map[string]interface{}
+	Valid  bool
+	Name   string
+}
+
+type hash map[string]interface{}
+
+func (event *Event) GetString(key string) (string, error) {
```
Comment (roeepolegfiverr) [inline worker/rabbit.go]: What about more complex messages? with hashes..

**Pull request closed without merging**